### PR TITLE
initial merge to master

### PR DESCRIPTION
**Pull request by mbyvcm**

### CODE REVIEW

File: CoverageCalculatorPy.py [hunk 1]
```diff
@@ -65,46 +82,87 @@ def main(args):
...
+    # iterate over each line of the bed file 
     with open(args.bedfile) as bed:
 
-        cnt_bed_ln = 1  # start on element 1 because 0 contains header
+        # counter used to access element of the features list that correpondes
+        # to a bedfile entry. Start on element 1 because 0 contains features header
+        cnt_bed_ln = 1
 
         for line in bed:
             bedlist = line.split()
```
Comment (josephhalstead) [inline CoverageCalculatorPy.py:168]: Line 177 - 
```
if len(bedlist) >2:
```

I think this should be?:
```
if len(bedlist) >3
```

Code won't run on my bed file without a meta column.

File: env/CoverageCalculatorPy.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - ncurses=6.1=hf484d3e_0
+  - openssl=1.1.1=h7b6447c_0
+  - pip=18.1=py36_0
+  - python=3.6.7=h0371630_0
+  - readline=7.0=h7b6447c_5
+  - setuptools=40.6.2=py36_0
+  - sqlite=3.25.3=h7b6447c_0
+  - tk=8.6.8=hbc83047_0
+  - wheel=0.32.3=py36_0
+  - xz=5.2.4=h14c3975_4
```
Comment (josephhalstead) [inline env/CoverageCalculatorPy.yml]: conda env won't load - try removing most of these conda depenencies or removing the hash e.g.h6b74fdf_2

This worked for me:
```
name: CoverageCalculatorPy
channels:
  - conda-forge
  - bioconda
  - defaults
dependencies:
  - python=3.6.7
  - zlib=1.2.11
  - pip:
    - pytabix==0.1

```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,42 @@
...
+
+### Tabix indexing a GATK3 DepthOfCoverage file
+
+The 'depth of coverage' file must be tabix indexed. The first three columns of the depthfile must be; chromosome, coordinate and depth. A file generated in GATK3 can be indexed as follows:
+
+```
+sed 's/:/\t/g' <GATK depthOfCoverage file> | grep -v 'Locus' | sort -k1,1 -k2,2n | bgzip > <filename.gz>
+
+tabix -b 2 -e 2 -s 1 <filename.gz> 
+```
```
Comment (josephhalstead) [inline README.md:42]: sed command on mac different:
```
sed "s/:/$(printf '\t')/g" <GATK depthOfCoverage file>  |  grep -v 'Locus' | sort -k1,1 -k2,2n | bgzip > <filename.gz>

```

**Comment by josephhalstead** [comment]
When I try and run on my file I get that a gap on the y chromosome between -1 and 0?

Will send you the coverage and bed for testing

**Review by josephhalstead — COMMENTED** [review]
Hi Chris,

Code looks good! All tests pass fine.

I spotted a few minor things which I have described below. I will send you the coverage file and bed file I had a problem with!

 - When I try and run on my file I get that a gap on the y chromosome between -1 and 0?

File: CoverageCalculatorPy.py (hunk 1 above)
Comment (mbyvcm) [reply CoverageCalculatorPy.py:168]: Correct. Your suggested change has been added.

File: README.md (hunk 3 above)
Comment (mbyvcm) [reply README.md:42]: added to readme

File: env/CoverageCalculatorPy.yml (hunk 2 above)
Comment (mbyvcm) [reply env/CoverageCalculatorPy.yml]: changes yml file as given above

**Comment by mbyvcm** [comment]
- code review complete
- raised issues resolved
- documentation complete
- tests passed
- validation to be performed as part of pipeline validation

**Pull request merged**